src/main/java/hudson/plugins/git/GitChangeLogParser.java

```diff
@@ -69,7 +71,9 @@ public GitChangeLogParser(GitClient git, boolean authorOrCommitter) {
     }
     
     public List<GitChangeSet> parse(@NonNull InputStream changelog) throws IOException {
-        return parse(IOUtils.readLines(changelog, "UTF-8"));
+        try (InputStreamReader isr = new InputStreamReader(changelog, StandardCharsets.UTF_8); LineIterator it = new LineIterator(isr)) {
+            return parse(it);
+        }
```

Comment:
Improved implementation: still uses Commons I/O, but now streams the lines rather than loading the whole file into memory.